firebase_admin/_sseclient.py

```diff
@@ -37,6 +40,35 @@ def rebuild_auth(self, prepared_request, response):
         pass
 
 
+class _EventBuffer(object):
+    """A helper class for buffering and parsing raw SSE data."""
+
+    def __init__(self):
+        self._buffer = []
+        self._tail = ''
+
+    def append(self, char):
+        self._buffer.append(char)
+        self._tail += char
+        if len(self._tail) > 4:
+            self._tail = self._tail[1:]
```

Comment:
You could also just do `self._tail = self._tail[-4:]` instead of the `if`.

Reply:
Done